### PR TITLE
Prompt user to save current board before auto-creating a board #1077

### LINKED ISSUE #1077: Prompt user to save current board before auto-creating a board
As of now, auto-creating a board will clear the panels then execute all panel adding then save it as a board. The problem is that user's current board will be cleared and it's not saved.

We should prompt user if they would like to save their current board before auto-creating a board.

### PR DESCRIPTION
**Pull request by m133225**
Fixes #1077

Just realised I didn't create a PR for this. Did it over 3 weeks ago :fearful: 
Will look through again before asking for review.

### CODE REVIEW

**Comment by m133225** [comment]
Ready for review. This is the dialog prompted before auto-creating a board.
screen shot 2016-03-30 at 2 29 52 am

File: src/main/java/ui/BoardAutoCreator.java [hunk 1]
```diff
@@ -59,20 +61,39 @@ public BoardAutoCreator(UI ui, PanelControl panelControl, Preferences prefs) {
...
 
         return autoCreate;
     }
 
+    private void saveBoardAfterUserConfirmation(String boardName) {
+        if (shouldSave(boardName)) ui.getMenuControl().saveBoardAs();
+    }
+
+    private boolean shouldSave(String boardName) {
+
```
Comment (acruis) [inline src/main/java/ui/BoardAutoCreator.java]: Extra line?

File: src/main/java/ui/BoardAutoCreator.java [hunk 2]
```diff
@@ -59,20 +61,39 @@ public BoardAutoCreator(UI ui, PanelControl panelControl, Preferences prefs) {
...
         autoCreate.getItems().add(workAllocation);
 
         return autoCreate;
     }
 
+    private void saveBoardAfterUserConfirmation(String boardName) {
+        if (shouldSave(boardName)) ui.getMenuControl().saveBoardAs();
+    }
+
+    private boolean shouldSave(String boardName) {
```
Comment (acruis) [inline src/main/java/ui/BoardAutoCreator.java]: Method name does not make it obvious that a dialog box will be created.

**Comment by acruis** [comment]
I feel there should also be a test which confirms that the current board is actually saved when the user clicks `Yes`.

**Comment by m133225** [comment]
@acruis Ready for review. Renamed the method and added the mentioned test!

File: src/test/java/guitests/BoardAutoCreatorTest.java [hunk 3]
```diff
@@ -37,13 +41,43 @@ public void cleanUpBoards() {
...
     @Test
-    public void milestoneBoardAutoCreationTest() {
+    public void boardAutoCreator_clickYesInSavePrompt_currentBoardSaved() {
+        int panelCount = panelControl.getPanelCount();
+        assertEquals(0, panelControl.getNumberOfSavedBoards());
+
+        // create 3 new panels
+        pushKeys(new KeyCodeCombination(KeyCode.P, KeyCombination.SHORTCUT_DOWN));
+        pushKeys(new KeyCodeCombination(KeyCode.P, KeyCombination.SHORTCUT_DOWN));
+        pushKeys(new KeyCodeCombination(KeyCode.P, KeyCombination.SHORTCUT_DOWN));
```
Comment (acruis) [inline src/test/java/guitests/BoardAutoCreatorTest.java]: Use `KeyboardShortcuts.CREATE_RIGHT_PANEL`?

File: src/main/java/ui/BoardAutoCreator.java [hunk 4]
```diff
@@ -59,20 +61,38 @@ public BoardAutoCreator(UI ui, PanelControl panelControl, Preferences prefs) {
...
         autoCreate.getItems().add(workAllocation);
 
         return autoCreate;
     }
 
+    private void saveBoardAfterUserConfirmation(String boardName) {
+        if (promptUserIfShouldSave(boardName)) ui.getMenuControl().saveBoardAs();
+    }
+
+    private boolean promptUserIfShouldSave(String boardName) {
```
Comment (damithc) [inline src/main/java/ui/BoardAutoCreator.java]: `isSaveBoardDialogResponsePositive` may read better in the call site. e.g. `if(isSaveBoardDialogResponsePositive(..)) ...`

**Comment by m133225** [comment]
@acruis Made the appropriate changes!

**Comment by acruis** [comment]
No more comments from me.

**Pull request merged**